src/include/reconcile_inline.h

```diff
@@ -502,9 +502,12 @@ __wt_rec_time_window_clear_obsolete(
     if (!tw->prepare && !F_ISSET(S2C(session), WT_CONN_IN_MEMORY)) {
         /*
          * Check if the start of the time window is globally visible, and if so remove unnecessary
-         * values.
+         * values. Avoid doing anything if the values are already clear, as setting the cleared flag
+         * can cause row-store reconciliation to do extra work repacking values.
          */
-        if (WT_REC_TW_START_VISIBLE_ALL(r, tw)) {
```

Comment:
Does the check above WT_TIME_WINDOW_HAS_START do something similar? Isn't it?

Reply:
Hmm. I guess so -- not sure how I missed that.

I think that means this change is useless...

Reply:
Yes.

Reply:
Earlier there was another macro to check whether the time window exists or not. As we are clearing only the start time window here, I changed the macro. Maybe you might have checked the code before I did that change.

Reply:
No, I think I just missed it. Happens; that's why we do reviews :-)

thanks for catching it

Reply:
(and sorry about the noise)